lua/entities/emplacements_turret_base/shared.lua

```diff
@@ -67,73 +95,70 @@ function ENT:Use( plr )
         if call == false then return end
 
         if IsValid( plr.CurrentEmplacement ) then
-            if SERVER then
-                -- plays sound on self to remind players this is an intended feature
-                self:EmitSound( "common/wpn_denyselect.wav", 60 )
-            end
-
-            return
+            plr.CurrentEmplacement:EmplacementDisconnect() -- hotswap emplacements! feels much better than being denied
+            
         end
-
-        self:SetShooter( plr )
-        self:StartShooting()
-        self.ShooterLast = plr
+        
+        self:EmplacementConnect( plr )
+        
     else
         if plr == self.Shooter then
-            self:SetShooter( nil )
-            self:FinishShooting()
+            self:EmplacementDisconnect()
         end
     end
 end
 
 function ENT:Think()
-    if not IsValid( self.turretBase ) and SERVER then
-        SafeRemoveEntity( self )
+    if not IsValid( self.turretBase ) then
+        if SERVER then SafeRemoveEntity( self ) end
+        return nil
     else
-        if IsValid( self ) then
+        if not IsValid( self ) and IsValid( self.turretBase ) then return end
+        if SERVER then
+            self.BasePos = self.turretBase:GetPos()
+            self.OffsetPos = self.turretBase:GetAngles():Up()
```

Comment:
Could you set these in `PhysicsSimulate` instead? Seems better to set it up where you need it. Especially because PhysicsSimulations stop when the entity is frozen (I believe)

Reply:
Yes, good idea, emplacement bases are always owned by world and so basically never get frozen but it's good to future-proof, maybe in the next pr?